File: 541reverseStringII/reverseStringII.java
class Solution {
    public String reverseStr(String s, int k) {
        final List<String> result = s.codePoints().mapToObj($ -> String.format("%c", $)).collect(Collectors.toList());
        for (int $ = 0; $ < result.size(); $ += 2 * k) Collections.reverse(result.subList($, Math.min($ + k, result.size())));
        return result.stream().collect(Collectors.joining());
    }
}

File: 541reverseStringII/reverseStringII.cpp
class Solution {
public:
    string reverseStr(string s, int k) {
        for (std::size_t _{}; _ < std::size(s); _ += 2 * k) std::reverse(std::next(std::begin(s), _), std::min(std::next(std::begin(s), _ + k), std::end(s)));
        return s;
    }
};
